PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/helpers/springcloud/SpringCloudAppPropertyView.java

```diff
@@ -0,0 +1,681 @@
+    private static final String ENABLE_PERSISTENT_STORAGE_KEY = "enablePersistentStorage";
+    public static final String CPU = "cpu";
+    public static final String MEMORY_IN_GB_KEY = "memoryInGB";
+    public static final String JVM_OPTIONS_KEY = "jvmOptions";
+    public static final String JAVA_VERSION_KEY = "javaVersion";
+    public static final String ENABLE_TEXT = "Enable";
+    public static final String DISABLE_TEXT = "Disable";
+    public static final String EMPTY_TEXT = "Empty";
+
+    private JButton triggerPublicButton;
+    private JComboBox javaVersionCombo;
+    private JComboBox cpuCombo;
+    private JTextField jvmOpsTextField;
+    private JButton refreshButton;
+    private JButton startButton;
+    private JButton stopButton;
+    private JButton restartButton;
+    private JButton deleteButton;
+    private JLabel lblAppName;
+    private JLabel lblPublic;
+    private JPanel instanceDetailHolder;
+    private JPanel instanceDetailPanel;
+    private JPanel statusPanel;
+    private JButton triggerPersistentButton;
+    private JComboBox memCombo;
+    private JPanel mainPanel;
+    private JPanel publicPanel;
+    private JButton saveButton;
+    private JBTable instanceTable;
+    private HyperlinkLabel testUrlLink;
+    private HyperlinkLabel publicUrlHyperLink;
+    private JLabel subsLabel;
+    private JLabel resourceGroupLabel;
+    private JLabel clusterLabel;
+    private JLabel appNameLabel;
+    private JLabel persistentLabel;
+    private EnvironmentVariablesTextFieldWithBrowseButton envTable;
+    private HideableDecorator instancePanelDecorator;
+
+    private SpringAppViewModel model;
+    private Project project;
+    private AppResourceInner appResourceInner;
+    private DeploymentResourceInner deploymentResourceInner;
+    private String appId;
+    private String appName;
+    private DefaultTableModel instancesTableModel;
+
+    public SpringCloudAppPropertyView(Project project, String appId) {
+        this.project = project;
+        this.appId = appId;
+        this.appName = SpringCloudIdHelper.getAppName(appId);
+        instancesTableModel = new DefaultTableModel() {
+            public boolean isCellEditable(int var1, int var2) {
+                return false;
+            }
+        };
+        instancesTableModel.addColumn("App Instances Name");
+        instancesTableModel.addColumn("Status");
+        instancesTableModel.addColumn("Discover Status");
+        instanceTable.setModel(instancesTableModel);
+        instanceTable.setRowSelectionAllowed(true);
+        instanceTable.setSelectionMode(ListSelectionModel.SINGLE_SELECTION);
+        instanceTable.getEmptyText().setText("Loading instances status");
+        instanceTable.setPreferredSize(new Dimension(-1, 200));
+        this.saveButton.addActionListener(e -> {
+            wrapperOperations(TelemetryConstants.SAVE_SPRING_CLOUD_APP, project, (changes) -> {
+                if (changes.isEmpty()) {
```

Comment:
when will changes empty but save button enabled?

Reply:
There are some chances that dirty check is not performed on time.